platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -60,15 +88,48 @@ def get_presence(self):
         Returns:
             bool: True if PSU is present, False if not
         """
-        status = 0
-        if self.presence_file_exists:
-            try:
-                with open(os.path.join(self.psu_path, self.psu_presence), 'r') as presence_status:
-                    status = int(presence_status.read())
-            except (ValueError, IOError):
-                status = 0
+        if self.psu_presence is not None:
+            status = self._read_generic_file(os.path.join(self.psu_path, self.psu_presence), 0)
+            return status == 1
         else:
-            status = self.index in psu_list
+            return True
 
-        return status == 1
+    def get_voltage(self):
+        """
+        Retrieves current PSU voltage output
+
+        Returns:
+            A float number, the output voltage in volts, 
+            e.g. 12.1 
+        """
+        if self.psu_voltage is not None:
+            voltage = self._read_generic_file(os.path.join(self.psu_path, self.psu_voltage), 0)
+            return float(voltage) / 1000
+        else:
+            raise NotImplementedError
```

Comment:
I am not sure whether we should raise "NotImplemenetedError" here, considering one case, if there is some bug with sysFS and the symbolic link was deleted, in this case maybe return a "none" would be better? just a thought.

Reply:
it seems your suggestion is better. will fix it.